fix 'check_server' function in ntpclient's hotplug script

Before the server check function just examined the exit code of a single-run
of ntpclient.
But sadly ntpclient (according to the source) does not indicate closed ports
of the server with an exitcode > 0. Thus this patch verifies if the exit code
is zero _and_ if ntpclient prints an RFC 1305 compatible status output.

see https://dev.openwrt.org/ticket/15133

--- a/packages/net/ntpclient/files/ntpclient.hotplug
+++ b/packages/net/ntpclient/files/ntpclient.hotplug
@@ -28,7 +28,9 @@ check_server() {
 	}
 
 	[ -z "$hostname" ] && return
-	$NTPC -c 1 -p ${port:-123} -i 2 -h $hostname > /dev/null && { SERVER=$hostname; PORT=${port:-123}; }
+	# successful operation always results in an RFC-1305 Appendix A status message (digits, dots and spaces)
+	# connection failures are not supposed to raise errors (according to ntpclient's source)
+	"$NTPC" -c 1 -p ${port:-123} -i 2 -h "$hostname" | grep -q "^[0-9. \t]\+$" && { SERVER=$hostname; PORT=${port:-123}; }
 }
 
 set_drift() {
